src/segment-loader.js

```diff
@@ -354,66 +361,148 @@ export default class SegmentLoader extends videojs.EventTarget {
   }
 
   /**
-   * Determines what segment request should be made, given current
-   * playback state.
+   * The segment loader has no recourse except to fetch a segment in the
+   * current playlist and use the internal timestamps in that segment to
+   * generate a syncPoint. This function returns a good candidate index
+   * for that process.
+   *
+   * @param {Object} playlist - the playlist object to look for a
+   * @returns {Number} An index of a segment from the playlist to load
+   */
+  getSyncSegmentCandidate_(playlist) {
+    if (this.currentTimeline_ === -1) {
+      return 0;
+    }
+
+    let segmentIndexArray = playlist.segments
+      .map((s, i) => {
+        return {
+          timeline: s.timeline,
+          segmentIndex: i
+        };
+      }).filter(s => s.timeline === this.currentTimeline_);
+
+    if (segmentIndexArray.length) {
+      return segmentIndexArray[Math.min(segmentIndexArray.length - 1, 1)].segmentIndex;
```

Comment:
This looks like if there is only one segment with the current timeline, we pick that, otherwise pick the second of the list. Why are these options better candidates than any of the other segments?